### ISSUE TITLE
Add LGDO format conversion utilities

### ISSUE THREAD

**Issue by gipert** [legend-exp/legend-pydataobj#4]
We should implement a method for each LGDO to convert underlying data to third-party formats like NumPy, Pandas, AwkwardArray. I'm thinking about something like:
```py
lgdo_obj.convert(fmt="pandas.DataFrame", copy=False)
```
Where `fmt` could take `pandas.DataFrame`, `numpy.ndarray`, `awkward.Array`.

This way, we would store the conversion code along with the LGDO implementation and make it easier to jump between data representations (like in `load_nda()`, `load_pd()`, `build_tcm()`, the `DataLoader`, etc).

We need of course to make a distinction between copy and zero-copy conversions.

**Comment by MoritzNeuberger** [comment]
I am confused about how the return type annotation would work in this case. Can you have a single function with multiple types of output depending on the input parameters?

**Comment by gipert** [comment]
Yes, it would look like this:
```py
def convert(...) -> pandas.DataFrame | numpy.NDArray | ...:
    pass
```

**Comment by MoritzNeuberger** [comment]
Over the last few days, I have been playing around with implementing this feature. For the most part, it is straightforward, although a few questions arose:

VectorOfVectors:
- To convert it to a numpy.ndarray, I now first convert it to an aoesa using `to_aoesa` and use its `convert` function. `to_aoesa` also uses `np.empty` to implement the nda, and when `preserve_dtype` is set to `True` we also have the problem that the previously empty entries are filled with random values. I assume it is not preferable to have `preserve_dtype` set to `False`.

Struct/Table: 
- The implementation in numpy.ndarray is not easy either. For now, I solved it by returning a dict containing the key and value entries of Struct/Table in two separate numpy arrays. What would be a better way to implement this?

WaveformTable/encoded data: 
- Does it need `convert`?

`copy`:
- I have implemented this option wherever possible. That is, always for pandas.DataFrame and when necessary for numpy.ndarrays. AFAIK awkward arrays usually do not copy?

ToDos:
- [ ] Figure out how to implement units with pint. Is this possible with awkward arrays?
- [ ] Write tests
